### PR TITLE
v1.12.0

### PR DESCRIPTION
**Pull request by IvanPleshkov**
# Qdrant v1.12 API updates

# New APIs examples

```csharp
using Qdrant.Client;
using Qdrant.Client.Grpc;

var client = new QdrantClient("localhost", 6334);

await client.SearchMatrixOffsetsAsync(
	"collection",
	filter: MatchKeyword("bar", "hello"),
	sample: 3,
    	limit: 2
);
```

```csharp
using Qdrant.Client;
using Qdrant.Client.Grpc;

var client = new QdrantClient("localhost", 6334);

await client.SearchMatrixPairsAsync(
	"collection",
	filter: MatchKeyword("bar", "hello"),
	sample: 3,
    	limit: 2
);
```

```csharp
using Qdrant.Client;
using Qdrant.Client.Grpc;

var client = new QdrantClient("localhost", 6334);

await client.FacetAsync(
	"collection_1",
	filter: MatchKeyword("bar", "hello"),
	key: "foo",
	limit: 2
);
```

# TODO

- [ ] bump version after container release

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <PackageProjectUrl>https://github.com/qdrant/qdrant-dotnet</PackageProjectUrl>
     <PackageReleaseNotes>https://github.com/qdrant/qdrant-dotnet/releases</PackageReleaseNotes>
     <PackageTags>qdrant, database, vector, search</PackageTags>
-    <QdrantVersion>v1.11.0</QdrantVersion>
+    <QdrantVersion>dev</QdrantVersion>
```
Comment (IvanPleshkov) [inline Directory.Build.props:12]: TODO: remove after release

File: tests/Qdrant.Client.Tests/PointTests.cs [hunk 2]
```diff
@@ -556,5 +556,44 @@ await _client.UpsertAsync("collection_1", new[]
...
+			fieldName: "foo",
+			schemaType: PayloadSchemaType.Keyword
+		);
+
+		await _client.FacetAsync(
+			"collection_1",
+			filter: MatchKeyword("bar", "hello"),
+			key: "foo",
+    		limit: 2
+		);
```
Comment (russcam) [inline tests/Qdrant.Client.Tests/PointTests.cs:575]: should this assert something about the response?

File: tests/Qdrant.Client.Tests/PointTests.cs [hunk 3]
```diff
@@ -556,5 +556,44 @@ await _client.UpsertAsync("collection_1", new[]
...
+			sample: 3,
+    		limit: 2
+		);
+
+		await _client.SearchMatrixOffsetsAsync(
+			"collection_1",
+			filter: MatchKeyword("bar", "hello"),
+			sample: 3,
+    		limit: 2
+		);
```
Comment (russcam) [inline tests/Qdrant.Client.Tests/PointTests.cs:595]: should this assert something about the response?

File: tests/Qdrant.Client.Tests/PointTests.cs [hunk 4]
```diff
@@ -556,5 +556,44 @@ await _client.UpsertAsync("collection_1", new[]
...
+	public async Task SearchMatrix()
+	{
+		await CreateAndSeedCollection("collection_1");
+
+		await _client.SearchMatrixPairsAsync(
+			"collection_1",
+			filter: MatchKeyword("bar", "hello"),
+			sample: 3,
+    		limit: 2
+		);
```
Comment (timvisee) [inline tests/Qdrant.Client.Tests/PointTests.cs:588]:
```suggestion
		await _client.SearchMatrixPairsAsync(
			"collection_1",
			filter: MatchKeyword("bar", "hello"),
			sample: 3,
			limit: 2
		);
```

File: tests/Qdrant.Client.Tests/PointTests.cs (hunk 3 above)
Comment (timvisee) [inline tests/Qdrant.Client.Tests/PointTests.cs:595]:
```suggestion
		await _client.SearchMatrixOffsetsAsync(
			"collection_1",
			filter: MatchKeyword("bar", "hello"),
			sample: 3,
			limit: 2
		);
```

File: tests/Qdrant.Client.Tests/PointTests.cs (hunk 2 above)
Comment (timvisee) [inline tests/Qdrant.Client.Tests/PointTests.cs:575]:
```suggestion
		await _client.FacetAsync(
			"collection_1",
			filter: MatchKeyword("bar", "hello"),
			key: "foo",
			limit: 2
		);
```

Comment (IvanPleshkov) [reply tests/Qdrant.Client.Tests/PointTests.cs]: This test checks only API validity. I used it to test snippets in PR description which will be included in the documentation. Do you think it's unnecessary?

File: tests/Qdrant.Client.Tests/PointTests.cs (hunk 3 above)
Comment (IvanPleshkov) [reply tests/Qdrant.Client.Tests/PointTests.cs]: This test checks only API validity. I used it to test snippets in PR description which will be included in the documentation. Do you think it's unnecessary?

File: Directory.Build.props (hunk 1 above)
Comment (timvisee) [reply Directory.Build.props:12]:
```suggestion
    <QdrantVersion>v1.12.0</QdrantVersion>
```

**Pull request merged**